rllib/algorithms/sac/torch/sac_torch_learner.py

```diff
@@ -314,14 +310,23 @@ def compute_gradients(
             for component in (
                 ["qf", "policy", "alpha"] + ["qf_twin"] if config.twin_q else []
             ):
-                self.metrics.peek((module_id, component + "_loss")).backward(
+                # Get the optimizer for the current component and module.
+                optim = self.get_optimizer(module_id, component)
+                # Zero the gradients. Note, we need to reset the gradients b/c
+                # each component for a module operates on the same graph.
+                optim.zero_grad(set_to_none=True)
+                # Compute the gradients for the component and module.
+                self.metrics.peek(module_id, component + "_loss").backward(
                     retain_graph=True
                 )
+                # Store the gradients for the component and module.
+                # TODO (simon): Check another time the graph for overlapping
+                # gradients.
                 grads.update(
                     {
-                        pid: p.grad
+                        pid: p.grad.clone()
```

Comment:
Can you explain why we need to `clone` this?

Reply:
Add a comment with the explanation.

Maybe this is the same problem I have in torch DreamerV3 ...

Reply:
This merely a safeguard to avoid the gradients stored for certain parameters being inadvertedly changed by the optimizer in further operations.